File: contact-list-ui/src/service/ContactListService.js
import axios from 'axios';

const HOST = 'http://localhost:8080'
const CONTACT_LIST_BY_PAGE = HOST+'/contactLists';
const CONTACT_LIST_BY_NAME = HOST+'/contactName/';

class ContactListService{
   contactListByPage(page){
        return  axios.get(CONTACT_LIST_BY_PAGE+`?page=${page}&size=10`);
    }
    contactListByName(name){
        return  axios.get(CONTACT_LIST_BY_NAME+`${name}`);
    }
}
export default new ContactListService();

File: contactlist/src/main/java/com/learning/contactlist/service/ContactListService.java
package com.learning.contactlist.service;

import com.learning.contactlist.exception.ContactNameNotFoundException;
import com.learning.contactlist.model.ContactListModel;
import com.learning.contactlist.repo.ContactListModelRepo;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class ContactListService {

    @Autowired
    ContactListModelRepo contactListModelRepo;

    public Page<ContactListModel> findAllContactLists(Pageable pageable) {
        return contactListModelRepo.findAll(pageable);
    }

    public List<ContactListModel> findContactByName(String contactName) throws ContactNameNotFoundException {
        List<ContactListModel> contactByName =  contactListModelRepo.findByContactName(contactName);
        if(contactByName.isEmpty()){
            throw new ContactNameNotFoundException(contactName+" name not found");
        }
        return contactByName;
    }
}
